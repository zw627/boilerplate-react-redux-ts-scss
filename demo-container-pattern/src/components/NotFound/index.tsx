import React from "react";
import { Link } from "react-router-dom";

const NotFound: React.FC<Record<string, unknown>> = () => {
  return (
    <div className="not-found">
      <h1>Error 404</h1>
      <p>Not Found</p>
      <Link to="/">Back to Home</Link>
    </div>
  );
};

export default NotFound;
