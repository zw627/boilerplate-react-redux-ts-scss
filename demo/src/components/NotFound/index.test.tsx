import React from "react";
import { BrowserRouter } from "react-router-dom";
import { mount, ReactWrapper } from "enzyme";

import NotFound from "Components/NotFound";

describe("NotFound", () => {
  let wrapper: ReactWrapper;

  // Setup wrapper
  beforeEach(() => {
    wrapper = mount(
      <BrowserRouter>
        <NotFound />
      </BrowserRouter>
    ) as ReactWrapper;
  });

  test("compare to the last snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
