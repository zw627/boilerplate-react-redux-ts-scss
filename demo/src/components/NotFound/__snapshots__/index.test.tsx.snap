// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound compare to the last snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <NotFound>
      <div
        className="not-found"
      >
        <h1>
          Error 404
        </h1>
        <p>
          Not Found
        </p>
      </div>
    </NotFound>
  </Router>
</BrowserRouter>
`;
